react-components/tests/tests-utilities/fixtures/sdk.ts

```diff
@@ -31,7 +31,12 @@ export const postMock = vi.fn<
     options?: HttpRequestOptions
   ) => Promise<HttpResponse<ExternalIdsResultList<AssetProperties>>>
 >(async (_path, options) => {
-  const assetRef = options?.data.items[0] as DmsUniqueIdentifier;
+  const assetRef = (options?.data as { items: DmsUniqueIdentifier[] | undefined } | undefined)
```

Comment:
This is pretty ugly - why is it needed?

Reply:
It's added to fix a type error. In general, this is a problematic case - we're trying to mock a very general method (`post`), but only want to handle types that are specific to the relevant test. So we have to do some casting to ensure we can read the data properly from the input request body. In this case, the mocking should probably be happening in the test file itself, and not this overarching SDK-mock file :thinking: But I think that's out of scope for this PR.